src/main/java/org/folio/linked/data/service/search/InstanceSearchServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package org.folio.linked.data.service.search;
+
+import static java.lang.String.format;
+import static java.util.Objects.isNull;
+import static java.util.stream.Collectors.joining;
+
+import java.util.Collection;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.folio.linked.data.client.SearchClient;
+import org.folio.linked.data.domain.dto.SearchResponseTotalOnly;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class InstanceSearchServiceImpl implements InstanceSearchService {
+
+  private static final String EXCLUDE_SUPPRESSED = "staffSuppress <> \"true\" and discoverySuppress <> \"true\"";
+  private static final String ID_NOT_EQUALS = "id <> \"%s\"";
+  private static final String LCCN_EQUALS = "lccn==\"%s\"";
+
+  private final SearchClient searchClient;
+
+  @Override
+  public SearchResponseTotalOnly searchByLccn(Collection<String> lccn) {
+    return search(getLccnQuery(lccn));
+  }
+
+  @Override
+  public SearchResponseTotalOnly searchByLccnExcludingId(Collection<String> lccn, String instanceId) {
+    if (isNull(instanceId)) {
+      return searchByLccn(lccn);
+    }
+    return search(format("%s and %s", getLccnQuery(lccn), format(ID_NOT_EQUALS, instanceId)));
+  }
+
+  private SearchResponseTotalOnly search(String query) {
+    try {
+      return searchClient.searchInstances(query).getBody();
+    } catch (Exception e) {
+      throw new RuntimeException("Unable to reach search service");
```

Comment:
Throwing generic exception is a bad practice. Why don't we throw failedDependencyException right here?

Reply:
Because search service is dependency for validation and might be a dependency for other components. If we throw failedDependencyException right here we couldn't know for which component dependency is failed, e.g. 
LccnValidation ->  InstancesSearch (search throws exception and we can say to user that validation failed because of search, otherwise we could only say 'Search is unavailable' without reference to validation which might be confusing)

Reply:
Replaced RuntimeException with SearchException

Reply:
ended up removing the exception throwing.